packages/api/app/auth.js

```diff
@@ -51,6 +51,11 @@ async function ensureAuthorized(req, res, next) {
     return next();
   } catch (error) {
     log.error(error);
+    if (launchpadProtectedAuth()
```

Comment:
So when a Launchpad token is passed to an endpoint as a bearer token, it will necessarily fail JWT verification and then hit this code block which verifies it against Launchpad? Thus allowing people to use tokens obtained directly from Launchpad?

Reply:
So this PR is so that the dashboard can use SAML authentication.  The launchpadSaml.js builds a (what I'm calling) a saml-based JWT, and when we protect the cumulus API with Launchpad, it needs to accept a sm_token (directly from curling the Launchpad endpoint outside of cumulus), but also a JWT from the dashboard after SAML authentication.

Reply:
But the workflow is as you described.  If it's a JWT (from dashboard) do your thing, if it's malformed see if it might be a sm_token.

Reply:
Yeah I just trying to think through if there was any possible hole in this where invalid tokens could make it through. However, since we're falling back to checking the token against Launchpad, I don't think there is